test/integration/dialects/postgres/schema.test.js

```diff
@@ -0,0 +1,116 @@
+'use strict';
+
+const chai = require('chai');
+
+const expect = chai.expect;
+const Support = require('../../support');
+
+const dialect = Support.getTestDialect();
+const { DataTypes, Sequelize } = require('@sequelize/core');
+const { Config: config } = require('../../../config/config');
+const { sequelize } = require('../../support');
+
+if (dialect.startsWith('postgres')) {
+
+  describe('[POSTGRES Specific] Schema', () => {
+    beforeEach(async () => {
+      await sequelize.createSchema('testschema');
+    });
+
+    afterEach(async () => {
+      await sequelize.dropSchema('testschema');
+    });
+
+    describe('Permissions', () => {
+      /** create a new user that has restricted privileges, isolated to schema */
+      const restricted = {
+        username: 'restricted_user',
+        password: 'restricted_password',
+      };
+
+      beforeEach(async () => {
+        // create a new user that will have restricted privileges
+        await sequelize.query(`DROP USER IF EXISTS ${restricted.username};`);
+        await sequelize.query(`CREATE USER ${restricted.username} WITH PASSWORD '${restricted.password}';`);
+        await sequelize.query(`REVOKE CREATE ON SCHEMA public FROM ${restricted.username};`);
+        await sequelize.query(`GRANT ALL PRIVILEGES ON SCHEMA testschema TO ${restricted.username};`);
+
+        // connect restricted user to database
+        this.restrictedConnection = await new Sequelize(
+          config[dialect].database,
+          restricted.username,
+          restricted.password,
+          {
+            host: config[dialect].host,
+            port: config[dialect].port,
+            dialect,
+            transactionType: 'IMMEDIATE',
+          },
+        );
+        this.restrictedConnection.options.quoteIdentifiers = true;
+
+        // query interfaces
+        this.queryInterface = await sequelize.getQueryInterface();
+        this.restrictedUserQI = await this.restrictedConnection.getQueryInterface();
+      });
+
+      afterEach(async () => {
+        const query = await sequelize.getQueryInterface();
+        await query.dropTable({ schema: 'public',     tableName: 'my_test_table' });
+        await query.dropTable({ schema: 'testschema', tableName: 'my_test_table' });
+        await query.dropTable({ schema: 'public',     tableName: 'shouldnt_create' });
+        await query.dropTable({ schema: 'testschema', tableName: 'shouldnt_create' });
+        await sequelize.dropSchema('testschema');
+        await sequelize.query(`DROP USER IF EXISTS ${restricted.username};`);
+      });
+
+      it.skip('prevents schema-privileged user from creating public tables', async () => {
+        expect(() => {
+          this.restrictedUserQI.createTable(
+            'shouldnt_create',
+            { name: DataTypes.STRING },
+            { schema: 'public' },
+          );
+        }).to.throw(); // or to.be.rejected
```

Comment:
Per your comments on Slack, I've updated this and feel so much better that these no longer have to be skipped :) The lines can be improved.  I'm using `rejectedWith(Error)`, but wish I knew what error so it could be more explicit.

Reply:
You can pass a string to check a message, I agree it would be nice to be more explicit. If you put a dummy string it should print what error message is used in your console when you run the test. You can then copy it here :)